tests/functional/src/main/java/org/mule/functional/classloading/isolation/classloader/MuleClassLoaderRunnerFactory.java

```diff
@@ -0,0 +1,265 @@
+import org.mule.runtime.module.artifact.classloader.ArtifactClassLoader;
+import org.mule.runtime.module.artifact.classloader.ArtifactClassLoaderFilter;
+import org.mule.runtime.module.artifact.classloader.ArtifactClassLoaderFilterFactory;
+import org.mule.runtime.module.artifact.classloader.ClassLoaderFilter;
+import org.mule.runtime.module.artifact.classloader.ClassLoaderLookupPolicy;
+import org.mule.runtime.module.artifact.classloader.CompositeClassLoader;
+import org.mule.runtime.module.artifact.classloader.DefaultArtifactClassLoaderFilterFactory;
+import org.mule.runtime.module.artifact.classloader.FilteringArtifactClassLoader;
+import org.mule.runtime.module.artifact.classloader.MuleArtifactClassLoader;
+import org.mule.runtime.module.artifact.classloader.MuleClassLoaderLookupPolicy;
+import org.mule.runtime.module.extension.internal.manager.DefaultExtensionManager;
+
+import com.google.common.collect.Sets;
+
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Factory that creates that mimics the class loading hierarchy used in a standalone container.
+ * <p/>
+ * The class loaders created have the following hierarchy:
+ * <p/>
+ * <ul>
+ * <li>Container: all the provided scope dependencies plus their dependencies (if they are not test) and java</li>
+ * <li>Plugins (optional): for each plugin a class loader will be created with all the compile scope dependencies and their dependencies (only the ones with scope compile)</li>
+ * <li>Application: all the test scope dependencies and their dependencies if they are not defined to be excluded, plus the test dependencies
+ * from the compile scope dependencies (again if they are not excluded).</li>
+ * </ul>
+ *
+ * @since 4.0
+ */
+public class MuleClassLoaderRunnerFactory
+{
+
+    protected final transient Logger logger = LoggerFactory.getLogger(this.getClass());
+    private ArtifactClassLoaderFilterFactory artifactClassLoaderFilterFactory = new DefaultArtifactClassLoaderFilterFactory();
+    private DefaultExtensionManager extensionManager = new DefaultExtensionManager();
+
+    /**
+     * Creates a {@link ClassLoaderTestRunner} containing the container, plugins and application {@link ArtifactClassLoader}s
+     * @param klass
+     * @param artifactUrlClassification
+     * @return a {@link ClassLoaderTestRunner}
+     */
+    public ClassLoaderTestRunner createClassLoader(Class<?> klass, ArtifactUrlClassification artifactUrlClassification)
+    {
+        final TestContainerClassLoaderFactory testContainerClassLoaderFactory = new TestContainerClassLoaderFactory(getExtraBootPackages(klass), artifactUrlClassification.getContainerURLs().toArray(new URL[0]));
```

Comment:
Klass params is used only to get the extraBootPackages. Instead of passing the class, pass that extra boot packages